### ISSUE TITLE
Can't login to phpMyAdmin while setting REQUIRE SSL

### ISSUE THREAD

**Issue by lusu007** [phpmyadmin/docker#272]
Regarding to our conversation [here ](https://github.com/phpmyadmin/docker/issues/142)I open this issue.

I'm trying to login to my phpMyAdmin instance with a user using ```REQUIRE SSL```, but I get an error message as the result. The error message can be seen in the included Screenshot.
![image](https://user-images.githubusercontent.com/10341994/75615499-b7562000-5b44-11ea-9d12-646bf41d0855.png)
****

The login with SSL via MySQLWorkbench or the console works just fine. Even with ```REQUIRE SSL``` enabled.

I'm using the latest version of the official phpMyAdmin Docker image from DockerHub.
I configured phpMyAdmin with a MariaDB instance inside a ```docker-compose.yml```.
```yml
version: '3.7'
services:
  mariadb:
    image: mariadb:latest
    container_name: mariadb
    restart: always
    command: ['--wait_timeout=28800']
    env_file: .env
    environment:
      - MYSQL_ROOT_PASSWORD=${PASSWORD}
    volumes:
      - mariadb_data:/var/lib/mysql
      - ./config/mariadb:/etc/mysql/conf.d
      - ./config/server-certs:/etc/certs
    networks:
      legend-data:
        ipv4_address: 10.0.4.123
  phpmyadmin:
    image: phpmyadmin/phpmyadmin:latest
    container_name: phpmyadmin
    restart: always
    env_file: .env
    environment:
      - MYSQL_ROOT_PASSWORD=${PASSWORD}
      - PMA_HOST=mariadb
    volumes:
      - ./config/phpmyadmin/config.user.inc.php:/etc/phpmyadmin/config.user.inc.php
      - ./config/client-certs:/etc/client-certs
    networks:
      legend-data: null
      legend-web:
        ipv4_address: "10.0.1.151"
volumes:
  mariadb_data:
networks:
  legend-data:
    external:
      name: legend-data
  legend-web:
    external:
      name: legend-web
```

I configured the SSL part for phpMyAdmin in a custom ```config.user.inc.php``` file.
```php
$i = 1;
$cfg['Servers'][$i]['ssl'] = true;
$cfg['Servers'][$i]['ssl_key'] = '/etc/client-certs/client-key.pem';
$cfg['Servers'][$i]['ssl_cert'] = '/etc/client-certs/client-cert.pem';
$cfg['Servers'][$i]['ssl_ca'] = '/etc/client-certs/ca.pem';
$cfg['Servers'][$i]['ssl_ca_path'] = '/etc/client-certs/';

$cfg['Servers'][$i]['auth_type'] = 'config';
$cfg['Servers'][$i]['user'] = 'test';
$cfg['Servers'][$i]['password'] = '12345';
```

phpMyAdmin sits behind an nginx reverse proxy. Over this reverse proxy I can access phpMyAdmin via HTTPS.

Furthermore I modified my MariaDB config slightly, as it is described on DockerHub, to implement SSL certificates.
```ini
[mysqld]
# ssl certificates and tls
tls_version = TLSv1.3
ssl-ca=/etc/certs/ca.pem
ssl-cert=/etc/certs/server-cert.pem
ssl-key=/etc/certs/server-key.pem
```

If some information are missing please let me know.

**Comment by williamdes** [comment]
Thank you !
Time to sleep, I will have a look ASAP

**Comment by lusu007** [comment]
How is it going here? Did you find a mistake or a solution?

**Comment by shyblower** [comment]
Same problem here although I configured
$cfg['Servers'][1]['ssl'] = true;
in /etc/phpmyadmin/config.user.inc.php

**Comment by williamdes** [comment]
Sorry for the delay :O 

Adding this to my TODO list of the week

**Comment by williamdes** [comment]
Hi @lusu007 !

I had 0 issues with my setup using docker
Will try another time soon.

You will find my example here: 
https://github.com/wdesportes/phpmyadmin-local-setup/blob/a9ac9dc281bcfa3348e4c7ad10e00ef71cbcf546/docker-compose.yml#L23

Feel free to browse the files on the repository, you will be able to find all the context

**Comment by williamdes** [comment]
Everything okay ?
I could not reproduce this issue

**Comment by lusu007** [comment]
Hey @williamdes ,

first of all sorry for my late anwser. 
We ditched the SSL support in our project. So we do not need it anymore.

But I will have a look in you repository. Thank you :)

**Comment by shyblower** [comment]
@williamdes, are your mariadb and mysql instances built with openssl or some other ssl implementation?

**Comment by williamdes** [comment]
> @williamdes, are your mariadb and mysql server instances built with openssl or some other ssl implementation?

I did not build the docker images, they are the original/official ones from MySQL and MariaDB :)

**Comment by shyblower** [comment]
@williamdes, has the DB user you are connecting with from phpmyadmin to your DBs, been created with the "REQUIRE SSL" flag?

**Comment by williamdes** [comment]
Yes, I did it using the Web UI and it worked

**Comment by darootler** [comment]
I am still not able to use phpmyadmin with ssl.

Could anyone please point me in the right direction?

Regards

**Comment by williamdes** [comment]
@darootler Could you give us the phpMyAdmin version you are using ?
And if possible a step by step description so I can try to reproduce this issue on my side

**Comment by darootler** [comment]
Here is the docker compose file i am using:

```
version: '3'

services:
  db:
    environment:
      MYSQL_ROOT_PASSWORD: XXX
    image: mariadb:rc
    ports:
      - "5711:3306"
    restart: always
    volumes:
      - /etc/stunnel/backup.cert:/etc/stunnel/backup.cert
      - /etc/stunnel/backup.key:/etc/stunnel/backup.key
      - /etc/stunnel/uca.pem:/etc/stunnel/uca.pem
      - /share/CACHEDEV1_DATA/Container/volumes/mariadb/etc/mysql/conf.d:/etc/mysql/conf.d:ro
      - /share/CACHEDEV1_DATA/Container/volumes/mariadb/var/lib/mysql:/var/lib/mysql
      
  dbtool:
    depends_on:
      - db
    environment:
      PMA_HOST: db
      PMA_PORT: 3306
    image: phpmyadmin/phpmyadmin:latest
    ports:
      - "4711:80"
    restart: always
    volumes:
      - /share/CACHEDEV1_DATA/Container/volumes/phpmyadmin/etc/phpmyadmin/config.user.inc.php:/etc/phpmyadmin/config.user.inc.php
```

config.user.inc.php:

`<?php
$i = 1;
$cfg['Servers'][$i]['ssl'] = true;
`

I am getting the same error message as the issue creater.

Regards
Richard

**Comment by williamdes** [comment]
@darootler does the mysql command line connect successfully ?

**Comment by williamdes** [comment]
I hope this issue is solved on your side, on mine everything works :)

**Comment by pr8kerl** [comment]
For anyone with similar issue, I hit this when trying to connect to an AWS rds instance using phpmyadmin 5.1 container image.
I discovered the rds mysql engine was 5.6 which only supports TLSv1.0. However ubuntu image of phpmyadmin has min TLS version set to TLSv1.2.
To workaround I updated last two lines of /etc/ssl/openssl.cnf to...
```
[system_default_sect]
MinProtocol = TLSv1.0
CipherString = DEFAULT@SECLEVEL=1
```
There's prob a better way than setting this globally at the OS level.
